ua-snap/data-api
# Permafrost CSV exports implemented, metadata added to CSVs, and CSV 404 checks working.

Closes #47.

This PR depends on the `csv_improvements` branch of the iem-webapp.

To test, run the data-api like this:

```
git checkout csvs_for_iem_webapp
export WEB_APP_URL=http://localhost:3000/
pipenv run flask run
```

And run the iem-webapp like this:

```
git checkout csv_improvements
export SNAP_API_URL=http://localhost:5000
npm run dev
```

Visit the iem-web app at http://localhost:3000 and load a report for each of the following:
- A lat/lon point (not a pre-defined community)
- A community
- A HUC
- A protected area
- A fire management unit
- A corporation
- A climate division
- An ethnolinguistic region

For each of the types of report above, verify that there is a CSV download button for each of the following types of data (if they are displayed on the report page):

- Temperature
- Precipitation
- Permafrost
- Flammability
- Vegetation change

Load each of the CSV downloads into a spreadsheet application and verify that the metadata, columns, and values make sense. Also verify that the filename of each CSV file is named after the corresponding place name.

I did some research and there appears to be no standard way to add metadata/comments to a CSV file. Currently, metadata is added as lines starting with "#" at the beginning of the CSV.
